moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -1096,6 +1096,10 @@ bool TimeOptimalTrajectoryGeneration::doTimeParameterizationCalculations(robot_t
 
     if (diverse_point)
       points.push_back(new_point);
+    // If the last point is not a diverse_point we replace the last added point with it to make sure to always have the
```

Comment:
I don't have anything against the fix. But why is the variable called `diverse_point` when it seems to mean, based on the comments, simply "unique waypoint", which would be easier to understand?

Reply:
> I don't really understand the fix. If it's not a diverse_point, that means it's identical to the previous point, right?

It's identical with some tolerance `min_angle_change_`, it was causing [MTC to fail](https://github.com/ros-planning/moveit_task_constructor/blob/master/core/src/stages/connect.cpp#L127-L132) because the last waypoint after parameterizing the trajectory isn't the expected goal waypoint

Reply:
OK, that makes sense